nixos/modules/security/tpm2.nix

```diff
@@ -0,0 +1,120 @@
+{ lib, pkgs, config, ... }:
+let
+  cfg = config.security.tpm2;
+
+  # This snippet is taken from tpm2-tss/dist/tpm-udev.rules, but modified to allow custom user/groups
+  # The idea is that the tssUser is allowed to acess the TPM and kernel TPM resource manager, while
+  # the tssGroup is only allowed to access the kernel resource manager
+  # Therefore, if either of the two are null, the respective part isn't generated
+  udevRules = tssUser: tssGroup: ''
+    ${lib.optionalString (tssUser != null) ''KERNEL=="tpm[0-9]*", MODE="0660", OWNER="${tssUser}"''}
```

Comment:
Since this function is only used once, I'd remove the arguments and use `cfg.tss{User,Group}` directly in it

Reply:
I think this function is complicated enough (or at least not very easy to read). Having user and group configurable, I'd _really_ like to insist on having this separate, even if it's just for reducing the visual complexity of the config section.